File: SudokuKMP/shared/src/commonMain/kotlin/com/nextjedi/sudokustreak/model/Difficulty.kt
package com.nextjedi.sudokustreak.model

enum class Difficulty(val label: String, val clues: Int, val cellsToRemove: Int) {
    EASY("Easy", 46, 35),
    MEDIUM("Medium", 36, 45),
    HARD("Hard", 26, 55)
}

File: SudokuKMP/iosApp/iosApp/Models/Difficulty.swift
import Foundation

enum Difficulty: String, CaseIterable, Identifiable {
    case easy = "Easy"
    case medium = "Medium"
    case hard = "Hard"

    var id: String { rawValue }

    var clues: Int {
        switch self {
        case .easy: return 46
        case .medium: return 36
        case .hard: return 26
        }
    }

    var cellsToRemove: Int {
        switch self {
        case .easy: return 35
        case .medium: return 45
        case .hard: return 55
        }
    }
}
